OneSignalDevelopers/ngosi
# Add page to view all attendees

Fixes #33 

## Solution

I added a page where a presenter can see all the attendees who've ever downloaded their slides. 

###### Groupies page
<img width="870" alt="Screen Shot 2021-12-27 at 2 27 53 PM" src="https://user-images.githubusercontent.com/1715082/147505139-cb9d5508-d194-41b4-b590-d2b5427a92be.png">

## Testing

To test this correctly, you will need to create at least two accounts that have completed onboarding (aka the user has a completed profile in the database) to test this properly.
 
1. Start Supabase CLI, `supabase start`
1. Create the view using the script in `db.sql`
2. Start the dev server, `yarn dev`
3. Log in to your account, [here](http://localhost:3000/)
3. Navigate to the [create preso page](http://localhost:3000/preso), fill out the form, and submit (you should be taken to the QR page if successful. If you get undefined in the URL bar, something failed, and you should leave a comment on the failure and repro steps.)
4. Navigate to the survey page and fill out the survey
5. Navigate to the [groupies page](http://localhost:3000/groupies)
6. Observe that there is one audience member on the list
7. Return to step 4 and log in with a different user account.

Note, you may have already had surveys and preso created, so it may be easier to compare what's changed instead of focusing on the numbers indicated in the testing steps.

## Resources

###### View creation script to run
https://github.com/OneSignalDevelopers/ngosi/blob/cc7c6e5cc49f949febf4d27fb7b0e7e96001bbcb/db.sql#L66-L78

###### Navigating to the Survey page from the QR page
![2021-12-27 14 19 11](https://user-images.githubusercontent.com/1715082/147504629-d8029792-88ec-4761-b286-c4b4e04e56ee.gif)


### References

- [`.in` filter docs](https://supabase.com/docs/reference/javascript/in)
- [Suggestion to use a view instead of using the supabase client to do joins](https://github.com/supabase/supabase/discussions/2438#discussioncomment-1697441)
